### PR TITLE
Correctly report errors thrown by CommandLinePathFactory#create.

### PR DESCRIPTION
**Pull request by tjgq**
This method is called while initializing the remote module [1]. Any exceptions
not derived from java.io.IOException cause a silent server crash.

[1] https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java;l=436;drc=f3211f00ae08746b5794ab01d404c32b43146aba

### CODE REVIEW

**Comment by tjgq** [comment]
@bazel-io fork 5.3.0

**Comment by tjgq** [comment]
@Yannic

File: src/test/java/com/google/devtools/build/lib/runtime/CommandLinePathFactoryTest.java [hunk 1]
```diff
@@ -99,6 +100,9 @@ public void createWithNamedRoot() throws Exception {
         .isEqualTo(filesystem.getPath("/path/to/output/base/foo"));
     assertThat(factory.create(ImmutableMap.of(), "%output_base%/foo/bar"))
         .isEqualTo(filesystem.getPath("/path/to/output/base/foo/bar"));
+
+    assertThat(factory.create(ImmutableMap.of(), "%workspace%//foo//bar"))
```
Comment (Yannic) [inline src/test/java/com/google/devtools/build/lib/runtime/CommandLinePathFactoryTest.java:104]: Why allow this?

**Comment by Yannic** [comment]
@bazel-io flag

File: src/test/java/com/google/devtools/build/lib/runtime/CommandLinePathFactoryTest.java (hunk 1 above)
Comment (tjgq) [reply src/test/java/com/google/devtools/build/lib/runtime/CommandLinePathFactoryTest.java:104]: It seems gratuitous to disallow double-slashes when most other command-line utilities handle them correctly; in addition, we get it for free through `PathFragment.create`. We could even avoid the edge case by converting the whole string to a `PathFragment` before regex-matching the first segment, but I wanted to do the bare minimum to avoid complicating the cherry pick.

Comment (Yannic) [reply src/test/java/com/google/devtools/build/lib/runtime/CommandLinePathFactoryTest.java:104]: Fair enough
